src/game/client/draw/w3dmodeldraw.cpp

```diff
@@ -13,3 +13,1149 @@
+        { "IgnoreConditionStates", &BitFlags<117ul>::Parse_From_INI, nullptr, offsetof(W3DModelDrawModuleData, m_ignoreConditionStates) },
+        { "ReceivesDynamicLights", &INI::Parse_Bool, nullptr, offsetof(W3DModelDrawModuleData, m_recievesDynamicLights) },
+        { nullptr, nullptr, nullptr, 0 },
+    };
+    // clang-format on
+
+    ModuleData::Build_Field_Parse(p);
+    p.Add(dataFieldParse, 0);
+}
+
+void W3DModelDrawModuleData::Xfer_Snapshot(Xfer *xfer)
+{
+    uint8_t ver = 1;
+    xfer->xferVersion(&ver, 1);
+
+    for (auto &i : m_conditionStates) {
+        xfer->xferByte((int8_t *)&i.m_validStuff);
+#ifdef GAME_DEBUG_STRUCTS
+        xfer->xferAsciiString(&i.m_description);
+#endif
+
+        if (i.m_validStuff) {
+            for (auto &j : i.m_boneMap) {
+                xfer->xferInt(&j.second.index);
+                xfer->xferUser((void *)&j.second.transform, sizeof(j.second.transform));
+            }
+
+            for (int j = 0; j < 2; j++) {
+                xfer->xferInt(&i.m_turretInfo[j].m_turretAngleBone);
+                xfer->xferInt(&i.m_turretInfo[j].m_turretPitchBone);
+            }
+
+            for (int j = 0; j < BONE_COUNT; j++) {
+                for (auto &k : i.m_weaponBarrelInfoVec[j]) {
+                    xfer->xferUser((void *)&k.m_weaponLaunchBoneTransform, sizeof(k.m_weaponLaunchBoneTransform));
+                }
+            }
+        }
+    }
+}
+
+bool Does_State_Exist(std::vector<ModelConditionInfo> const &v, BitFlags<MODELCONDITION_COUNT> const &f)
+{
+    for (auto &i : v) {
+        for (int a = i.Get_Conditions_Count() - 1; a >= 0; a--) {
+            if (f == i.Get_Conditions_Yes(a)) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+uint64_t Make_Transition(int old_state, int new_state)
+{
+    return new_state | (uint64_t)((int64_t)old_state << 32);
+}
+
+void Parse_Bone_Name_Key(INI *ini, void *instance, void *store, void const *user_data)
+{
+    Utf8String str(ini->Get_Next_Token());
+    str.To_Lower();
+
+    if (instance) {
+        ((ModelConditionInfo *)instance)->Add_Public_Bone(str);
+    }
+
+    if (str.Is_Empty() || str.Is_None()) {
+        *(NameKeyType *)store = NAMEKEY_INVALID;
+    } else {
+        *(NameKeyType *)store = g_theNameKeyGenerator->Name_To_Key(str);
+    }
+}
+
+void Parse_Show_Hide_Sub_Object(INI *ini, void *instance, void *store, void const *user_data)
+{
+    std::vector<ModelConditionInfo::HideShowSubObjInfo> *v = (std::vector<ModelConditionInfo::HideShowSubObjInfo> *)store;
+    Utf8String str = ini->Get_Next_Ascii_String();
+    str.To_Lower();
+
+    if (str.Is_None()) {
+        v->clear();
+    } else {
+        while (str.Is_Not_Empty()) {
+            bool found = false;
+
+            for (auto &i : *v) {
+                if (!strcasecmp(i.name, str)) {
+                    i.visible = (uintptr_t)user_data != 0;
+                    found = true;

```

Comment:
Should there be a `break` here? It will keep iterating.

Reply:
Nope, the original didn't have one.